### ISSUE TITLE
Alert messages and User friendly messages not read by Talkback when any valid or invalid values entered in input fields.

### ISSUE THREAD

**Issue by A11yravi** [ampproject/amphtml#3937]
Short description of your issue:
Alert messages and User friendly messages not read by Talkback when any valid or invalid values entered in input fields.

Test URL: https://amp-mk-1.herokuapp.com/examples.build/forms.amp.max.html

Pre-requisite: Turn on Talkback from Settings >> Accessibility >>Talkback

How do we reproduce the issue?

Steps to reproduce:
1. Logon to Nexus 9 and open chrome browser
2. Open test URL provided above
3. Linear navigate till your name editbox and double tap.

Case 1:
1. Input valid values in name field and leave email field blank and double tap on subscribe button.

Actual results: 
Alert Message "Please fill out this field" is not getting read.
User friendly messages such as “ There are some input that needs fixing. Please fix them” not getting read.

Expected Result:
Messages should be read by talkback.

Case 2:
1. Input valid values in email field and leave name field blank and double tap on subscribe button.

Actual results: 
Alert Message "Please fill out this field" is not getting read.
User friendly messages such as “ There are some input that needs fixing. Please fix them” not getting read.

Expected Result:
Messages should be read by talkback.

Case 3:
1. Input valid values in name field and invalid values in email field as (xyzabc.com don't include "@") and double tap on subscribe button.

Actual results: 
Alert Message "Please include an "@" in the email address. 'xyzabc.com is missing an '@' " is not getting read.
User friendly messages such as “ There are some input that needs fixing. Please fix them” not getting read.

Expected Result:
Messages should be read by talkback.

Case 4:
1. Input valid values in name and email field and double tap on subscribe button.

Actual results: 
User friendly messages such as “ Everything looks good, you can submit now ” not getting read.

Expected Result: 
Message should be read by talkback.

What browsers are affected?

OS : Android 6.0
Device: Nexus 9 (tablet)
Browser: Chrome (Version 51.0.2704.81)

Please find attached snapshot for reference.
![amp form error message 1](https://cloud.githubusercontent.com/assets/20249603/16636807/3f7ca408-438f-11e6-8c20-b5e723685010.png)
![amp form error message 2](https://cloud.githubusercontent.com/assets/20249603/16636805/3f77926a-438f-11e6-9967-baaef9d7c486.png)
![amp form error message 3](https://cloud.githubusercontent.com/assets/20249603/16636806/3f7a6daa-438f-11e6-81f7-2e1d327190b2.png)
![amp form valid message 4](https://cloud.githubusercontent.com/assets/20249603/16636813/46c0da7c-438f-11e6-98ca-a1134b4c5511.png)

**Comment by mkhatib** [comment]
So it looks like this might be a Chrome accessibility issue. At least for the validation bubbles part. I couldn't get the built-in bubbles to be read out by `VoiceOver` on Mac.

As for the other friendly messages, these are in the example using CSS and classes hooks. AMP is not actually displaying these the publisher just using the classes we're adding to the form to display these messages. We don't currently provide a hook for publishers to tell us which messages are displayed when the form becomes valid or when the form becomes invalid. 

We might add these when we get to work on custom validation UI and messages. But for now there's nothing AMP can do in this case.

**Comment by adelinamart** [comment]
@mkhatib any updates here? Thanks

**Comment by mkhatib** [comment]
Closing this. We've fixed a18y issues in forms messages.
